Bearded.UI/Controls/Control.cs

```diff
@@ -116,12 +141,18 @@ private void recalculateFrame()
             FrameChanged();
         }
 
-        public void RemoveFromParent() => Parent.Remove(this);
+        public void RemoveFromParent()
+        {
+            if (Parent == null)
+                throw new InvalidOperationException("This control doesn't have a parent.");
+
+            Parent.Remove(this);
+        }
 
         internal void AddTo(IControlParent parent)
         {
             if (Parent != null)
-                throw new InvalidOperationException();
+                throw new InvalidOperationException("This control already has a parent.");
```

Comment:
Do we actually need this check on this internal method which we have full control over? I'd rather make that a Debug.Assert perhaps?

Also ping @tomrijnbeek for how we want to handle cases like this in general.

Reply:
Oh haha, the check is already there, I only added a message to make it more clear why it's throwing. I won't revert these ones if you don't mind.